cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/internal/BootstrapCoordinator.scala

```diff
@@ -250,7 +252,23 @@ private[akka] final class BootstrapCoordinator(discovery: ServiceDiscovery,
 
   private def discoverContactPoints(): Unit = {
     log.info("Looking up [{}]", lookup)
-    discovery.lookup(lookup, settings.contactPointDiscovery.resolveTimeout).pipeTo(self)
+    discovery
+      .lookup(lookup, settings.contactPointDiscovery.resolveTimeout)
+      .map { resolved =>
+        if (lookup.portName.isDefined)
+          resolved
+        else if (resolved.addresses.flatMap(_.port).isEmpty)
+          Resolved(
+            resolved.serviceName,
+            resolved.addresses.map(target => ResolvedTarget(target.host, Some(defaultManagementPort), target.address))
+          )
+        else
+          Resolved(
+            resolved.serviceName,
+            resolved.addresses.filter(_.port.contains(defaultManagementPort))
+          )
+      }
+      .pipeTo(self)
```

Comment:
I think it could be a lot simpler if the fallback logic happens at the point of probing. In fact, it looks like there is already some kind of partially-implemented fallback logic there:

https://github.com/akka/akka-management/blob/9c97f6aa830db1da5f5fdff7febe07d7883b00a9/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/internal/BootstrapCoordinator.scala#L292

https://github.com/akka/akka-management/blob/23bc509c6d45df9c26d0b1b8f67bf9bf92846d95/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrapSettings.scala#L125-L126

https://github.com/akka/akka-management/blob/03a30b1b06c1dbc0e3e3522ee28872bec6e8b4ec/cluster-bootstrap/src/main/resources/reference.conf#L97-L101